### DISCUSSION TITLE
quickstart error

### DISCUSSION THREAD

**Discussion by azhsmesos** [GreptimeTeam/greptimedb#554] [Q&A] [answered]
<img width="1318" alt="image" src="https://user-images.githubusercontent.com/65518067/202361553-957243fb-aeb6-4485-80b5-ee6a63a4fe56.png">

run the `cargo run -- standalone start` command to report an error

**Comment by waynexia** [comment]
What toolchain are you using? Could please paste the output of `rustup show`? (here is mine)

```text
active toolchain
----------------

nightly-2022-07-14-aarch64-apple-darwin (overridden by '/Users/ruihang/repo/greptimedb/rust-toolchain')
rustc 1.64.0-nightly (87588a2af 2022-07-13)
```

**Reply by azhsmesos** [reply]
```
active toolchain
----------------

stable-x86_64-apple-darwin (directory override for '/Users/zhaozhenhang/Desktop/test/rust/greptimedb')
rustc 1.65.0 (897e37553 2022-11-02)

```

**Reply by azhsmesos** [reply]
It need nightly to build？

**Reply by waynexia** [reply]
Yes, we depend on some nightly features

**Reply by waynexia** [reply]
It looks strange, I see the "directory override" but it doesn't change the toolchain. A workaround is install this [toolchain](https://github.com/GreptimeTeam/greptimedb/blob/develop/rust-toolchain) manually by `default nightly-2022-07-14`

**Reply by azhsmesos** [reply]
Now, I used nightly and still reported error
```rust
   Compiling arrow2 v0.10.1
error[E0599]: no method named `lanes_eq` found for struct `std::simd::Simd` in the current scope
  --> /Users/zhaozhenhang/.cargo/registry/src/github.com-1ecc6299db9ec823/arrow2-0.10.1/src/types/simd/packed.rs:77:21
   |
77 |     (idx & vecmask).lanes_eq(idx)
   |                     ^^^^^^^^ help: there is a method with a similar name: `lanes`

error[E0599]: no method named `lanes_eq` found for struct `std::simd::Simd` in the current scope
  --> /Users/zhaozhenhang/.cargo/registry/src/github.com-1ecc6299db9ec823/arrow2-0.10.1/src/types/simd/packed.rs:87:21
   |
87 |     (idx & vecmask).lanes_eq(idx)
   |                     ^^^^^^^^ help: there is a method with a similar name: `lanes`

error[E0599]: no method named `lanes_eq` found for struct `std::simd::Simd` in the current scope
   --> /Users/zhaozhenhang/.cargo/registry/src/github.com-1ecc6299db9ec823/arrow2-0.10.1/src/types/simd/packed.rs:112:29
    |
112 |     (idx & left & vecmask1).lanes_eq(idx) | (idx & right & vecmask2).lanes_eq(idx)
    |                             ^^^^^^^^ help: there is a method with a similar name: `lanes`

error[E0599]: no method named `lanes_eq` found for struct `std::simd::Simd` in the current scope
   --> /Users/zhaozhenhang/.cargo/registry/src/github.com-1ecc6299db9ec823/arrow2-0.10.1/src/types/simd/packed.rs:112:70

```

**Reply by azhsmesos** [reply]
By the way, we have a wechat group or other communication tools?

**Reply by waynexia** [reply]
No, wechat is not ready. We current discuss on https://greptime.com/slack , welcome to join!

>Now, I used nightly and still reported error

To confirm, your current `rustup show` or `cargo -V` outputs the expected toolchain version, right?

**Reply by azhsmesos** [reply]
I'm not sure what to expect toolchain version, but my local project can be built successfully

**Reply by waynexia** [reply]
> I'm not sure what to expect toolchain version, but my local project can be built successfully

It should be `2022-07-13` or `2022-07-14`. Our project requires a specific rust version (the nightly api are not stable)

**Comment by evenyag** [comment]
Looks like you are using a stable channel. Could you plz provide your output of `cargo -V`?

**Reply by azhsmesos** [reply]
this `cargo 1.65.0 (4bc8f24d3 2022-10-20)`

**Reply by evenyag** [reply]
It looks like our `rust-toolchain` doesn't take effect.
https://github.com/GreptimeTeam/greptimedb/blob/bf408e3b96c8bd8b79b3931fc451a4ac62e16de0/rust-toolchain#L1

Have you ever installed some rust packages using tools like `homebrew`? It might overwrite your cargo path and cause some strange behavior. I met a similar problem recently after I installed `nextest` via homebrew. I fixed that by uninstalling `nextest` and reinstalling Rust.

**Reply by azhsmesos** [reply]
Yes, I installed something like protobuf via homebrew， but it was the startup tutorial that made me do it...

**Reply by evenyag** [reply]
`protobuf` doesn't matter, it isn't a Rust project. Could you execute `cargo -V` again, looks like your toolchain is still on stable. If the `rust-toolchain` takes effect, the output should look like this:
```
$ cargo -V
cargo 1.64.0-nightly (b1dd22e66 2022-07-09)
```

**Reply by azhsmesos** [reply]
It doesn't seem to be working. My date is 2022-11-11
```rust
zhaozhenhang@zhaozhenhangdeMacBook-Pro greptimedb % cargo -V
cargo 1.67.0-nightly (a3dfea71c 2022-11-11)
zhaozhenhang@zhaozhenhangdeMacBook-Pro greptimedb % rustc --version
rustc 1.67.0-nightly (a00f8ba7f 2022-11-15)
```

**Reply by evenyag** [reply]
What does `which cargo` output?

If you don't mind, try `brew uninstall rust` and use this command from Rust's [installation guide](https://www.rust-lang.org/tools/install)
```
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```

**Reply by azhsmesos** [reply]
input `which cargo` is ```rust
zhaozhenhang@zhaozhenhangdeMacBook-Pro greptimedb % which cargo
/Users/zhaozhenhang/.cargo/bin/cargo
```
I have performed the uninstallation and re-installation according to your requirements, but it still error when I builds.

**Reply by baoyachi** [reply]
Try use `cargo -Vv`

**Reply by evenyag** [reply]
Open a new terminal session? You could check the toolchain version using `cargo -V`, we need a specific nightly version to build the project. You need to ensure `cargo -V` outputs the version (`1.64.0-nightly (b1dd22e66 2022-07-09)`) in `rust-toolchain`.

**Comment by azhsmesos** [comment, marked as answer]
1.`rustup install nightly-2022-07-14`
2.`rustup override add nightly-2022-07-14`
That would solve the problem，I suggest you put it in quickstart docs.

**Reply by waynexia** [reply]
Thanks!
